contracts/cw-proposal-multiple/src/state.rs

```diff
@@ -0,0 +1,141 @@
+
+pub const MAX_NUM_CHOICES: u32 = 10;
+const NONE_OPTION_DESCRIPTION: &str = "None of the above";
+
+/// The governance module's configuration.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Config {
+    /// The threshold a proposal must reach to complete.
+    pub voting_strategy: VotingStrategy,
+    /// The minimum amount of time a proposal must be open before
+    /// passing. A proposal may fail before this amount of time has
+    /// elapsed, but it will not pass. This can be useful for
+    /// preventing governance attacks wherein an attacker aquires a
+    /// large number of tokens and forces a proposal through.
+    pub min_voting_period: Option<Duration>,
+    /// The default maximum amount of time a proposal may be voted on
+    /// before expiring.
+    pub max_voting_period: Duration,
+    /// If set to true only members may execute passed
+    /// proposals. Otherwise, any address may execute a passed
+    /// proposal.
+    pub only_members_execute: bool,
+    /// The address of the DAO that this governance module is
+    /// associated with.
+    pub dao: Addr,
+    /// Information about the depost required to create a
+    /// proposal. None if no deposit is required, Some otherwise.
+    pub deposit_info: Option<CheckedDepositInfo>,
+}
+
+/// Information about a vote that was cast.
+#[derive(Serialize, Deserialize, Clone, PartialEq, Eq, JsonSchema, Debug)]
+pub struct VoteInfo {
+    /// The address that voted.
+    pub voter: Addr,
+    /// Position on the vote.
+    pub vote: MultipleChoiceVote,
+    /// The voting power behind the vote.
+    pub power: Uint128,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+pub enum MultipleChoiceOptionType {
+    /// Choice that represents selecting none of the options; still counts toward quorum
+    /// and allows proposals with all bad options to be voted against.
+    None,
+    Standard,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct MultipleChoiceOptions {
+    pub options: Vec<MultipleChoiceOption>,
+}
+
+pub struct CheckedMultipleChoiceOptions {
+    pub options: Vec<CheckedMultipleChoiceOption>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct MultipleChoiceOption {
+    pub description: String,
+    pub msgs: Option<Vec<CosmosMsg<Empty>>>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct CheckedMultipleChoiceOption {
+    // This is the index of the option in both the vote_weights and proposal.choices vectors.
+    // Workaround due to not being able to use HashMaps in Cosmwasm.
+    pub index: u32,
+    pub option_type: MultipleChoiceOptionType,
+    pub description: String,
+    pub msgs: Option<Vec<CosmosMsg<Empty>>>,
+    pub vote_count: Uint128,
+}
+
+impl MultipleChoiceOptions {
+    pub fn into_checked(self) -> Result<CheckedMultipleChoiceOptions, ContractError> {
```

Comment:
is there tests for this function as an individual unit?

Reply:
added!